### PR TITLE
Feature/issue32

### PR DESCRIPTION
**Pull request by RobertK66**
This is ready to be pulled to devolp but I will wait for @kodsurf  to decide on the thruster branch to be pulled before .....

It sends a Hearbeat on the Debug Uart every 10 seconds. ( @AndyHilftnix: is this ok? )

I used a special module and event id so there are no 'unprintable' ASCII characters in this event binary:

image

### CODE REVIEW

File: src/radtest/radtest.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+void rtst_main (void){
+	if (RtstTick) {
+		// Called once every second.
+		RtstTick = false;
+		RtstTickCnt++;
+
+		if ((RtstTickCnt % RTST_HEARTBEAT_TICKS) == 0) {
+			// Make heartbeat every 10 sec.
+			SysEvent(RTST_MODNR, EVENT_WARNING, RTST_EVENTID_HEARTBEAT, "Supervision watchdog feed\n", 26);
```
Comment (kodsurf) [inline src/radtest/radtest.c:53]: Do I understand correctly that after this line code would be executed every 10s.

Thus sesor readout for LPC_I2C1 should be placed somewhere here ?

File: src/radtest/radtest.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#include "../mod/ado_timers.h"
+#include <ado_modules.h>
+
+#define 			RTST_MODNR					0x55			// ('U')  Module number for rad tests only
+#define 			RTST_EVENTID_HEARTBEAT		0x55			// ('U')  Event ID for heartbeat event
+
+#define 			RTST_TICK_MS				1000			// IRQ every second.
+#define 			RTST_HEARTBEAT_TICKS		  10			// Heartbeat every 10 seconds
+#define				RTST_MEMTST_TICKS			  16			// Memorytest all 15 seconds
+
```
Comment (kodsurf) [inline src/radtest/radtest.c:19]: Ok,

I will do RTST_SENSOR_TICKS 10

File: src/radtest/radtest.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	if (RtstTick) {
+		// Called once every second.
+		RtstTick = false;
+		RtstTickCnt++;
+
+		if ((RtstTickCnt % RTST_HEARTBEAT_TICKS) == 0) {
+			// Make heartbeat every 10 sec.
+			SysEvent(RTST_MODNR, EVENT_WARNING, RTST_EVENTID_HEARTBEAT, "Supervision watchdog feed\n", 26);
+		}
+		if ((RtstTickCnt % RTST_MEMTST_TICKS) == 0) {
```
Comment (kodsurf) [inline src/radtest/radtest.c:55]: and put sensor ticks here as 

if ((RtstTickCnt % RTST_SENSOR_TICKS) == 0) {
// read sensor from LPC_I2C1  and print i2c rx buffer  with sys_event() to debug uart
}

File: src/radtest/radtest.c [hunk 4]
```diff
@@ -55,6 +56,9 @@ void rtst_main (void){
 		if ((RtstTickCnt % RTST_MEMTST_TICKS) == 0) {
 			// Memory tests TODO
 		}
+		if ((RtstTickCnt % RTST_SENSOR_TICKS) == 0) {
+			SenReadAllValues();
```
Comment (kodsurf) [inline src/radtest/radtest.c:60]: image


Please check if this was indended for ISSUE-33

File: src/radtest/radtest.c [hunk 5]
```diff
@@ -51,7 +51,8 @@ void rtst_main (void){
 
 		if ((RtstTickCnt % RTST_HEARTBEAT_TICKS) == 0) {
 			// Make heartbeat every 10 sec.
-			SysEvent(RTST_MODNR, EVENT_WARNING, RTST_EVENTID_HEARTBEAT, "Supervision watchdog feed\n", 26);
+			//SysEvent(RTST_MODNR, EVENT_WARNING, RTST_EVENTID_HEARTBEAT, "Supervision watchdog feed\n", 26);
+			SysEvent_Debug("Supervision watchdog feed\n", 26)
```
Comment (kodsurf) [inline src/radtest/radtest.c:55]: I tried to make a function that prints pure strin without id message

Which is latter to be converted to pure binary string. 

image

I still have some ~ new line symbols comming from somewhere.

Please take a look. 

If no - we can revert this commit

**Comment by kodsurf** [comment]
I checked python code with Abbas. Looks like raspery pi also logs strings received over RS485.

Therefore I added basics of thruster module to feature/issue32 in preparation for rad test

image

Above screentshot shows "thr_hello" string with new line 0x0A printed. 

Python RS485 logger code looks for new line character and logs received line.

**Comment by kodsurf** [comment]
Pure binary string is now printed over debug uart

image


image


commits 
https://github.com/RobertK66/obc_1769_core/pull/36/commits/b9389638533861848d36dbf2dfa5e87fb746c3ab
and
https://github.com/RobertK66/obc_1769_core/pull/36/commits/dfeef619474c3541afb312dfe89f19331ad0741e

**Comment by kodsurf** [comment]
https://github.com/RobertK66/obc_1769_core/pull/36/commits/c136f45f409a7cc939c315d0eae9fbc41af431bd

This solves previous error with

image

**Comment by RobertK66** [comment]
Closing this because of merge mess. Will make new one from other branch....

**Pull request closed without merging**